### ISSUE TITLE
max_wait does not work correctly

### ISSUE THREAD

**Issue by scudette** [Velocidex/velociraptor#2818]
For event queries the client batches the results before sending to the lower of max_rows or max_wait (normally waiting for either 1000 rows or about 120 seconds).

However there is a bug here https://github.com/Velocidex/velociraptor/blob/11a548689cf71b4d67e6d6656500de67ea490424/actions/vql.go#L389

this is the code that batches the rows in memory before sending the payload
there are two conditions - if the row count exceeds the max_row then it sends the packet
and if the deadline is reached it sends the packet as well
but there is a bug where if a row is emitted the deadline is reset.
so in other words if the query returns a row the max_wait deadline is reset

**Comment by scudette** [comment]
Fixed by #2819
